### PR TITLE
basic implementation of update rest api

### PR DESCRIPTION
**Pull request by rajendersaini**
#3061 - update rest API basic implementation

### CODE REVIEW

**Comment by rajendersaini** [comment]
@bblommers  Please bless this PR

File: moto/apigateway/models.py [hunk 1]
```diff
@@ -618,13 +622,36 @@ def to_dict(self):
...
+        for op in patch_operations:
+            path = op["path"]
+            value = op["value"]
+            if op["op"] == "replace":
+                if "/name" in path:
+                    self.name = value
+                if "/description" in path:
+                    self.description = value
+                if "/apiKeySource" in path:
+                    if value not in ["HEADER", "AUTHORIZER"]:
```
Comment (bblommers) [inline moto/apigateway/models.py:645]: This check isn't necessary, as the validation already happens in responses.py

File: moto/apigateway/models.py [hunk 2]
```diff
@@ -905,6 +932,13 @@ def get_rest_api(self, function_id):
             raise RestAPINotFound()
         return rest_api
 
+    def update_rest_api(self, function_id, patch_operations):
+        rest_api = self.apis.get(function_id)
+        if rest_api is None:
+            raise RestAPINotFound()
```
Comment (bblommers) [inline moto/apigateway/models.py:938]: This edge case should have a unit test as well

**Review by bblommers — APPROVED** [review]
LGTM! Thanks again @rajendersaini
